mesonbuild/modules/python.py

```diff
@@ -497,7 +497,7 @@ def dependency_method(self, args: T.List['TYPE_var'], kwargs: 'TYPE_kwargs') ->
 
         # it's theoretically (though not practically) possible for the else clse
         # to not bind dep, let's ensure it is.
-        dep: 'Dependency' = NotFoundDependency(self.interpreter.environment)
+        dep: 'Dependency' = NotFoundDependency('python', self.interpreter.environment)
```

Comment:
We could be more precise here, in python_factory() it builds the full pkg-config name:
```
        pkg_embed = '-embed' if embed and mesonlib.version_compare(installation.version, '>=3.8') else ''
        pkg_name = f'python-{pkg_version}{pkg_embed}'
```

Reply:
```
pymod = import('python')
py_inst = pymod.find_installation('python3.9')
py_dep = py_inst.dependency()
message(py_dep.name())
```
This will print ```python```. So the name should be the same if the dependency was not found

I reverted this to the first version.